spec/takeaway_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'takeaway'
+
+describe Takeaway do
+  let(:order_class) { double "order_class" }
+  let(:text_client) { double "text_client" }
+  let(:takeaway) { described_class.new(order_class, text_client) }
+
+  before do
+    allow(STDIN).to receive(:gets).and_return("1111111111","1", "1", "", "yes")
+    allow(order_class).to receive(:show_menu)
+    allow(order_class).to receive(:dish_name).and_return("Fried Rice")
+    allow(order_class).to receive(:add_to_order).and_return({ "Fried Rice" => 1 })
+    allow(order_class).to receive(:item_list).and_return("1x Fried Rice £6.00\n")
+  end
+
+  it { is_expected.to respond_to :show_menu }
+  
+  it { is_expected.to respond_to :cost }
+
```

Comment:
Tests separated by method names which clearly show which aspect of the program is being tested in each block